src/MassiveActionHandler.ts

```diff
@@ -111,41 +117,38 @@ export class MassiveActionHandler extends AbstractActionHandler {
   }
 
   protected async handleWithState(handle: (state: any, context?: any) => void): Promise<void> {
-    await this.massiveInstance.withTransaction(async (tx: any) => {
-      let db
-      if (this.dbSchema === 'public') {
-        db = tx
-      } else {
-        db = tx[this.dbSchema]
-      }
-      this.warnOverwrite(db, 'migrate')
-      db.migrate = async (sequenceName: string) => await this.migrate(sequenceName, tx.instance)
-      this.warnOverwrite(db, 'txid')
-      db.txid = (await tx.instance.one('select txid_current()')).txid_current
+    const indexState = await this.loadIndexState()
```

Comment:
We should think about maybe using the cached `indexState` properties, same ones that are updated here: https://github.com/EOSIO/demux-js/blob/develop/src/AbstractActionHandler.ts#L331-L336

After closer inspection, we're inconsistent about when we use these cached values and when we use the `#loadIndexState` method. I think it might make sense to only use the cached values (perhaps replacing all the enumerated properties with an `indexState` property), which would be kept up-to-date whenever the database `indexState` changes, allowing less database reads and give a consistent, always-up-to-date way to access the data.